include/nbl/builtin/hlsl/glsl_compat/core.hlsl

```diff
@@ -15,6 +15,81 @@ namespace hlsl
 namespace glsl
 {
 
+template <typename T>
+T bitfieldInsert(T base, T insert, int32_t offset, int32_t bits)
+{
+    NBL_CONSTEXPR T one = typename unsigned_integer_of_size<sizeof(T)>::type(1);
+    const T mask = (one << bits) - one;
+    const T shifted_mask = mask << offset;
+
+    insert &= mask;
+    base &= (~shifted_mask);
+    base |= (insert << offset);
+
+    return base;
+}
```

Comment:
thats should be SPIR-V intrinsic, I beleive @Fletterio has implemented it already for Extract, follow that

Reply:
It's in the [workgroup fft branch](https://github.com/Devsh-Graphics-Programming/Nabla/blob/e98ef60dee98c2739802b4a95c9ea806bb35d3a9/include/nbl/builtin/hlsl/spirv_intrinsics/core.hlsl#L222). I can move that to another branch and push to master

Reply:
please do, thanks

Reply:
@Fletterio let me know when it is done

Reply:
Opened this PR but for some reason it also updated dxc

https://github.com/Devsh-Graphics-Programming/Nabla/pull/701

Alternatively I can just add only the spirv intrinsic + the function/struct and push straight to master

Reply:
> Opened this PR but for some reason it also updated dxc
> 
> #701
> 
> Alternatively I can just add only the spirv intrinsic + the function/struct and push straight to master

because you didn't update submodule after merging master, thats why and you committed a "changed" submodule pointer